---
pageClass: 'rule-details'
sidebarDepth: 0
title: 'svelte/no-inner-declarations'
description: 'disallow variable or `function` declarations in nested blocks'
since: 'v0.0.8'
---

# svelte/no-inner-declarations

> disallow variable or `function` declarations in nested blocks

- :gear: This rule is included in `"plugin:svelte/recommended"`.

## :book: Rule Details

This rule requires that function declarations and, optionally, variable declarations be in the root of a program or the body of a function.

This rule extends the base ESLint's [no-inner-declarations] rule. The AST generated by [svelte-eslint-parser] will false positives in [no-inner-declarations] rule because the root node of the script is not the `Program`.  
This rule supports [svelte-eslint-parser]'s AST.

[svelte-eslint-parser]: https://github.com/sveltejs/svelte-eslint-parser

<!--eslint-skip-->

```svelte
<script>
  /* eslint svelte/no-inner-declarations: "error" */

  /* ✓ GOOD */
  function doSomething() {}

  function doSomethingElse() {
    function doAnotherThing() {}
  }

  /* ✗ BAD */
  if (test) {
    function doSomethingBad() {}
  }
</script>
```

## :wrench: Options

```json
{
  "svelte/no-inner-declarations": [
    "error",
    "functions" // or "both"
  ]
}
```

Same as [no-inner-declarations] rule option. See [here](https://eslint.org/docs/rules/no-inner-declarations#options) for details.

## :couple: Related rules

- [no-inner-declarations]

[no-inner-declarations]: https://eslint.org/docs/rules/no-inner-declarations

## :rocket: Version

This rule was introduced in eslint-plugin-svelte v0.0.8

## :mag: Implementation

- [Rule source](https://github.com/sveltejs/eslint-plugin-svelte/blob/main/packages/eslint-plugin-svelte/src/rules/no-inner-declarations.ts)
- [Test source](https://github.com/sveltejs/eslint-plugin-svelte/blob/main/packages/eslint-plugin-svelte/tests/src/rules/no-inner-declarations.ts)

<sup>Taken with ❤️ [from ESLint core](https://eslint.org/docs/rules/no-inner-declarations)</sup>
