---
pageClass: 'rule-details'
sidebarDepth: 0
title: 'svelte/no-trailing-spaces'
description: 'disallow trailing whitespace at the end of lines'
since: 'v2.7.0'
---

# svelte/no-trailing-spaces

> disallow trailing whitespace at the end of lines

- :wrench: The `--fix` option on the [command line](https://eslint.org/docs/user-guide/command-line-interface#fixing-problems) can automatically fix some of the problems reported by this rule.

## :book: Rule Details

This rule extends the base ESLint's [no-trailing-spaces] rule. The [no-trailing-spaces] rule does not understand HTML comments and will report trailing whitespace in HTML comments when using `ignoreComments` option.
This rule supports HTML comments generated by [svelte-eslint-parser].

[svelte-eslint-parser]: https://github.com/sveltejs/svelte-eslint-parser

<!-- prettier-ignore-start -->
<!--eslint-skip-->

```svelte
<script>
  /* eslint svelte/no-trailing-spaces: "error" */

  /* ✓ GOOD */
  var foo = 0;
  /* ✗ BAD */
  var foo = 0;  
</script>

<!-- ✓ GOOD -->
<div>
  Text
</div>

<!-- ✗ BAD -->
<div>  
  Text  
</div>
```

<!-- prettier-ignore-end -->

## :wrench: Options

```jsonc
{
  "no-trailing-spaces": "off", // Don't need ESLint's no-trailing-spaces rule, so turn it off.
  "svelte/no-trailing-spaces": [
    "error",
    {
      "skipBlankLines": false,
      "ignoreComments": false
    }
  ]
}
```

- `skipBlankLines` ... If `true`, allows trailing whitespace on empty lines.
- `ignoreComments` ... If `true`, allows trailing whitespace in comments.

Same as [no-trailing-spaces] rule option. See [here](https://eslint.org/docs/rules/no-trailing-spaces#options) for details.

## :couple: Related rules

- [no-trailing-spaces]

[no-trailing-spaces]: https://eslint.org/docs/rules/no-trailing-spaces

## :rocket: Version

This rule was introduced in eslint-plugin-svelte v2.7.0

## :mag: Implementation

- [Rule source](https://github.com/sveltejs/eslint-plugin-svelte/blob/main/packages/eslint-plugin-svelte/src/rules/no-trailing-spaces.ts)
- [Test source](https://github.com/sveltejs/eslint-plugin-svelte/blob/main/packages/eslint-plugin-svelte/tests/src/rules/no-trailing-spaces.ts)

<sup>Taken with ❤️ [from ESLint core](https://eslint.org/docs/rules/no-trailing-spaces)</sup>
