---
pageClass: 'rule-details'
sidebarDepth: 0
title: 'svelte/no-unused-svelte-ignore'
description: 'disallow unused svelte-ignore comments'
since: 'v0.19.0'
---

# svelte/no-unused-svelte-ignore

> disallow unused svelte-ignore comments

- :gear: This rule is included in `"plugin:svelte/recommended"`.

## ğŸ“– Rule Details

This rule warns unnecessary `svelte-ignore` comments.

<ESLintCodeBlock>

<!--eslint-skip-->

```svelte
<script>
  /* eslint svelte/no-unused-svelte-ignore: "error" */
</script>

<!-- âœ“ GOOD -->
<!-- svelte-ignore a11y-autofocus a11y-missing-attribute -->
<img src="https://example.com/img.png" autofocus />

<!-- âœ— BAD -->
<!-- svelte-ignore a11y-autofocus a11y-missing-attribute -->
<img src="https://example.com/img.png" alt="Foo" />
```

</ESLintCodeBlock>

## ğŸ”§ Options

Nothing.

## ğŸ“š Further Reading

- [Svelte - Docs > Comments](https://svelte.dev/docs#template-syntax-comments)

## ğŸš€ Version

This rule was introduced in eslint-plugin-svelte v0.19.0

## ğŸ” Implementation

- [Rule source](https://github.com/sveltejs/eslint-plugin-svelte/blob/main/packages/eslint-plugin-svelte/src/rules/no-unused-svelte-ignore.ts)
- [Test source](https://github.com/sveltejs/eslint-plugin-svelte/blob/main/packages/eslint-plugin-svelte/tests/src/rules/no-unused-svelte-ignore.ts)
