# (svelte/prefer-writable-derived)

> description

## :book: Rule Details

This rule reports ???.

<!--eslint-skip-->

```svelte
<script>
  /* eslint svelte/prefer-writable-derived: "error" */
</script>

<!-- ✓ GOOD -->

<!-- ✗ BAD -->
```

## :wrench: Options

```json
{
  "svelte/prefer-writable-derived": ["error", {}]
}
```

-

## :books: Further Reading

-
